caffe2/CMakeLists.txt

```diff
@@ -211,10 +206,579 @@ if(NOT BUILD_ATEN_ONLY)
+    "${TORCH_SRC_DIR}/csrc/autograd/generated/python_variable_methods_dispatch.h"
+    "${TORCH_SRC_DIR}/csrc/autograd/generated/python_torch_functions_dispatch.h"
+    "${TORCH_SRC_DIR}/csrc/autograd/generated/python_nn_functions.h"
+    "${TORCH_SRC_DIR}/csrc/autograd/generated/python_nn_functions_dispatch.h"
+    )
+
+  set(GENERATED_THNN_SOURCES
+    ${GENERATED_THNN_CXX}
+    ${GENERATED_THNN_CXX_CUDA}
+    )
+
+  set(TORCH_GENERATED_CODE
+    ${GENERATED_CXX_TORCH}
+    ${GENERATED_THNN_SOURCES}
+    ${GENERATED_H_TORCH}
+    ${GENERATED_CXX_PYTHON}
+    ${GENERATED_H_PYTHON}
+    )
+
+  add_custom_command(
+    OUTPUT
+    ${TORCH_GENERATED_CODE}
+    COMMAND
+    "${PYTHON_EXECUTABLE}" tools/setup_helpers/generate_code.py
+      --declarations-path "${CMAKE_BINARY_DIR}/aten/src/ATen/Declarations.yaml"
+      --nn-path "aten/src"
+    DEPENDS
+    "${CMAKE_BINARY_DIR}/aten/src/ATen/Declarations.yaml"
+    "${CMAKE_CURRENT_LIST_DIR}/../aten/src/THNN/generic/THNN.h"
+    "${TOOLS_PATH}/autograd/templates/VariableType.h"
+    "${TOOLS_PATH}/autograd/templates/VariableType.cpp"
+    "${TOOLS_PATH}/autograd/templates/Functions.h"
+    "${TOOLS_PATH}/autograd/templates/Functions.cpp"
+    "${TOOLS_PATH}/autograd/templates/python_functions.h"
+    "${TOOLS_PATH}/autograd/templates/python_functions.cpp"
+    "${TOOLS_PATH}/autograd/templates/python_variable_methods.cpp"
+    "${TOOLS_PATH}/autograd/templates/python_variable_methods_dispatch.h"
+    "${TOOLS_PATH}/autograd/templates/python_torch_functions.cpp"
+    "${TOOLS_PATH}/autograd/templates/python_torch_functions_dispatch.h"
+    "${TOOLS_PATH}/autograd/templates/python_nn_functions.cpp"
+    "${TOOLS_PATH}/autograd/templates/python_nn_functions.h"
+    "${TOOLS_PATH}/autograd/templates/python_nn_functions_dispatch.h"
+    "${TOOLS_PATH}/autograd/templates/variable_factories.h"
+    "${TOOLS_PATH}/autograd/deprecated.yaml"
+    "${TOOLS_PATH}/autograd/derivatives.yaml"
+    "${TOOLS_PATH}/autograd/gen_autograd_functions.py"
+    "${TOOLS_PATH}/autograd/gen_autograd.py"
+    "${TOOLS_PATH}/autograd/gen_python_functions.py"
+    "${TOOLS_PATH}/autograd/gen_variable_factories.py"
+    "${TOOLS_PATH}/autograd/gen_variable_type.py"
+    "${TOOLS_PATH}/autograd/load_derivatives.py"
+    "${TOOLS_PATH}/autograd/nested_dict.py"
+    "${TOOLS_PATH}/autograd/utils.py"
+    "${TOOLS_PATH}/jit/gen_jit_dispatch.py"
+    "${TOOLS_PATH}/jit/templates/register_aten_ops.cpp"
+    WORKING_DIRECTORY "${TORCH_ROOT}")
+
+
+  # Required workaround for libtorch_python.so build
+  # see https://samthursfield.wordpress.com/2015/11/21/cmake-dependencies-between-targets-and-files-and-custom-commands/#custom-commands-in-different-directories
+  add_custom_target(
+    generate-torch-sources
+    DEPENDS ${TORCH_GENERATED_CODE}
+    )
+
+  
+  set(TORCH_SRCS
+    ${GENERATED_CXX_TORCH}
+    ${GENERATED_H_TORCH}
+    ${TORCH_SRC_DIR}/csrc/autograd/anomaly_mode.cpp
```

Comment:
it'd be nice to dedup this list with build_variables.py in the future (not this PR) - I wonder whether we can just read the same file in in both build systems